### ISSUE TITLE
Unexpected output with partially-`when`'d `switch`

### ISSUE THREAD

**Issue by bbrk24** [DanielXMoore/Civet#979]
```coffee
switch x
  when 1
    ;
  2
    ;
  else
    ;
```
```js
switch (x) {
  case 1:
    {
      break;
    }
    2;
  default: {
  }
}
```
I feel like that shouldn't be accepted.

**Comment by edemaine** [comment]
Yeah, we should probably detect a mixed switch and throw an error.

**Comment by STRd6** [comment]
I thought we detected it here: https://github.com/DanielXMoore/Civet/blob/0575162b63d2e912e7bc2f7583a0cc3c34d03d2d/source/lib.civet#L3196 but it appears we're missing a case.

**Comment by edemaine** [comment]
Actually, this seems to be a Playground issue; it's not respecting `"Error"` nodes. Will fix.

**Comment by bbrk24** [comment]
I originally noticed this because of a dead code warning in the LSP. If that's the case, does that mean the LSP is also not respecting "Error" nodes?

**Comment by edemaine** [comment]
Indeed, that seems to be the case. So now down to an LSP issue.

There should be a way to map these Errors to locations in the files, given where they appear in the AST. (This would be useful for LSP, CLI, and Playground.) 

@STRd6 if you have time to look at this, I won't for a while.

**Comment by STRd6** [comment]
This should be fixed by #993 and live in v0.3.12 of the LSP

The error message could be improved with better location information but that is a separate issue.

Thanks for the report and let me know if you see any other issues.
